app/scripts/background.js

```diff
@@ -439,6 +442,14 @@ function setupController(initState, initLangCode, remoteSourcePort) {
         // This ensures that UI is initialised only after background is ready
         // It fixes the issue of blank screen coming when extension is loaded, the issue is very frequent in MV3
         remotePort.postMessage({ name: 'CONNECTION_READY' });
+
+        // If we get a WORKER_KEEP_ALIVE message, we respond with an ACK
+        remotePort.onMessage.addListener((message) => {
+          if (message.name === WORKER_KEEP_ALIVE_MESSAGE) {
+            // To test un-comment this line and wait for 1 minute. An error should be shown on MetaMask UI.
+            remotePort.postMessage({ name: ACK_KEEP_ALIVE_MESSAGE });
+          }
+        });
```

Comment:
It is possible to rather use service worker for this message: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/app-init.js#L125

Reply:
Yup, the initial message uses SW, but at that point, we don't have a way to communicate back to the client (UI) as the streams have not been initialised.

So what I did was use a BroadcastChannel. 

I had a discussion with @danjm and we decided using remotePort/extensionPort would be better as we don't have to use another message channel to communicate with the UI